packages/material-ui-styles/src/getStylesCreator/getStylesCreator.js

```diff
@@ -50,12 +55,17 @@ export default function getStylesCreator(stylesOrCreator) {
               [
                 'Material-UI: You are trying to override a style that does not exist.',
                 `Fix the \`${key}\` key of \`theme.overrides.${name}\`.`,
+                'If you intentionally wanted to add new key, please use the theme.additions option',
```

Comment:
I think that it would be great to add a runtime check in case a prop is provided but doesn't map to any value in the theme.